src/pydvl/influence/frameworks/torch_differentiable.py

```diff
@@ -67,7 +72,8 @@ def solve_linear(
     matrix = model.hessian(
         all_x, all_y, progress=progress
     ) + hessian_perturbation * identity_tensor(model.num_params, device=model.device)
-    return torch.linalg.solve(matrix, b.T).T
+    info = {"hessian": matrix}
+    return torch.linalg.solve(matrix, b.T).T, info
```

Comment:
I think it would be better to change the interface to return a single dict with all the values instead of a tuple of tensor and dict.

Reply:
Or a named tuple .

Reply:
done